modules/dnn/src/net_impl.cpp

```diff
@@ -1830,15 +1830,270 @@ string Net::Impl::dump(bool forceAllocation) const
     return out.str();
 }
 
+static void dumpTensorToString(std::ostringstream &out, const Mat &m, const int num_indent_spaces = 4) {
+    string indent_spaces(num_indent_spaces, ' ');
+
+    int type = 1;
+    /* Check TensorProto::DataType from https://github.com/onnx/onnx/blob/main/onnx/onnx.proto */
+    switch (m.type()) {
+        case CV_32F: break;
+        case CV_8U:  type = 2; break;
+        case CV_8S:  type = 3; break;
+        case CV_16U: type = 4; break;
+        case CV_16S: type = 5; break;
+        case CV_32S: type = 6; break;
+/*
+#if OPENCV_VERSION_MAJOR > 4
+        case CV_64S: type = 7; break;
+        case CV_STR: type = 8; break;
+        case CV_BOOL: type = 9; break;
+#endif
+*/
```

Comment:
Please uncomment it.

Reply:
`OPENCV_VERSION_MAJOR` is undefined in `net_impl.cpp`. It is not going to work unless it is defined in 5.x.

Reply:
you just need to include "opencv2/core/version.hpp"

Reply:
The macro name is actually `CV_VERSION_MAJOR`, it is available already.